### PR TITLE
Prescheduling logic to controller

### PR DESCRIPTION
**Pull request by cwdsuzhou**
1. Add prescheduling status
2. Add filling occupied by filed

### CODE REVIEW

**Comment by cwdsuzhou** [comment]
ping @Huang-Wei  for review

File: pkg/controller/podgroup.go [hunk 1]
```diff
@@ -266,3 +276,17 @@ func (ctrl *PodGroupController) patchPodGroup(old, new *schedv1alpha1.PodGroup)
 	}
 	return nil
 }
+
+func fillOccupiedObj(pg *schedv1alpha1.PodGroup, pod *v1.Pod) {
+	refs := make([]string, 0)
```
Comment (Huang-Wei) [inline pkg/controller/podgroup.go:281]: nit: `var refs []string`

File: pkg/controller/podgroup.go [hunk 2]
```diff
@@ -266,3 +276,17 @@ func (ctrl *PodGroupController) patchPodGroup(old, new *schedv1alpha1.PodGroup)
...
+	for _, ownerRef := range pod.OwnerReferences {
+		refs = append(refs, fmt.Sprintf(pod.Namespace, ownerRef.Name))
```
Comment (Huang-Wei) [inline pkg/controller/podgroup.go:283]: "fmt.Sprintf(pod.Namespace, ownerRef.Name)" looks weird, I guess you want to use a `/` separator?

File: pkg/controller/podgroup.go [hunk 3]
```diff
@@ -266,3 +276,17 @@ func (ctrl *PodGroupController) patchPodGroup(old, new *schedv1alpha1.PodGroup)
...
+	}
+	if len(pg.Status.OccupiedBy) == 0 {
+		return
+	}
+	if len(refs) != 0 {
+		sortkeys.Strings(refs)
```
Comment (Huang-Wei) [inline pkg/controller/podgroup.go:289]: Is it equivalent to `sort.Strings(refs)`? If yes, let's eliminate the dependency on `sortkeys`.

File: pkg/controller/podgroup.go (hunk 2 above)
Comment (cwdsuzhou) [reply pkg/controller/podgroup.go:283]: Yes, sorry for mistake

File: pkg/controller/podgroup.go (hunk 3 above)
Comment (cwdsuzhou) [reply pkg/controller/podgroup.go]: removed

**Comment by cwdsuzhou** [comment]
@Huang-Wei  PR addressed

File: pkg/controller/podgroup.go [hunk 4]
```diff
@@ -266,3 +275,16 @@ func (ctrl *PodGroupController) patchPodGroup(old, new *schedv1alpha1.PodGroup)
...
+func fillOccupiedObj(pg *schedv1alpha1.PodGroup, pod *v1.Pod) {
+	var refs []string
+	for _, ownerRef := range pod.OwnerReferences {
+		refs = append(refs, fmt.Sprintf("%s/%s", pod.Namespace, ownerRef.Name))
+	}
+	if len(pg.Status.OccupiedBy) == 0 {
+		return
+	}
+	if len(refs) != 0 {
+		pg.Status.OccupiedBy = strings.Join(refs, ",")
```
Comment (Huang-Wei) [inline pkg/controller/podgroup.go:288]: Sort `refs`?

File: pkg/controller/podgroup.go (hunk 3 above)
Comment (Huang-Wei) [reply pkg/controller/podgroup.go]: I meant you can still sort them, but we can just use the default `sort` library.

Comment (cwdsuzhou) [reply pkg/controller/podgroup.go]: done

File: pkg/controller/podgroup.go (hunk 4 above)
Comment (cwdsuzhou) [reply pkg/controller/podgroup.go:290]: add sort.Strings()

**Comment by Huang-Wei** [comment]
@cwdsuzhou could you take a look at the flake? I created #99 for tracking.

This PR should be irrelevant.

/lgtm
/approve

**Pull request merged**